master/internal/webhooks/postgres_webhook.go

```diff
@@ -50,3 +74,238 @@ func DeleteWebhook(ctx context.Context, id WebhookID) error {
+		ws, err := workspace.WorkspaceByName(ctx, wName)
+		if err != nil {
+			return nil, err
+		}
+		w = ws
+
+		if w == nil {
+			return nil, fmt.Errorf("unable to find workspace with name: %v", wName)
+		}
+		wID = w.ID
+
+		pID, err := workspace.ProjectIDByName(ctx, wID, pName)
+		if pID != nil {
+			projectID = *pID
+		}
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if e.State == model.CompletedState {
+		status = "Your experiment completed successfully 🎉"
+		if webUIBaseURL != "" {
+			eURL = fmt.Sprintf("✅ <%v/det/experiments/%v/overview | %v (#%v)>",
+				webUIBaseURL, e.ID, e.Config.Name(), e.ID)
+		} else {
+			eURL = fmt.Sprintf("✅ %v (#%v)", e.Config.Name(), e.ID)
+		}
+		c = "#13B670"
+		mStatus = "Completed"
+	} else {
+		status = "Your experiment has stopped with errors"
+		if webUIBaseURL != "" {
+			eURL = fmt.Sprintf("❌ <%v/det/experiments/%v/overview | %v (#%v)>",
+				webUIBaseURL, e.ID, e.Config.Name(), e.ID)
+		} else {
+			eURL = fmt.Sprintf("❌ %v (#%v)", e.Config.Name(), e.ID)
+		}
+		c = "#DD5040"
+		mStatus = "Errored"
+	}
+	hours := e.EndTime.Sub(e.StartTime).Hours()
+	hours, m := math.Modf(hours)
+	minutes := int(m * 60)
+	duration := fmt.Sprintf("%vh %vmin", hours, minutes)
+	expBlockFields := []SlackField{
+		{
+			Type: "mrkdwn",
+			Text: fmt.Sprintf("*Status*: %v", mStatus),
+		},
+		{
+			Type: "mrkdwn",
+			Text: fmt.Sprintf("*Duration*: %v", duration),
+		},
+	}
+	if wID != 0 && wName != "" && webUIBaseURL != "" {
+		expBlockFields = append(expBlockFields, SlackField{
+			Type: "mrkdwn",
+			Text: fmt.Sprintf("*Workspace*: <%v/det/workspaces/%v/projects | %v>",
+				webUIBaseURL, wID, wName),
+		})
+	} else if wName != "" {
+		expBlockFields = append(expBlockFields, SlackField{
+			Type: "mrkdwn",
+			Text: fmt.Sprintf("*Workspace*: %v", wName),
+		})
+	}
+	if projectID != 0 && pName != "" && webUIBaseURL != "" {
+		expBlockFields = append(expBlockFields, SlackField{
+			Type: "mrkdwn",
+			Text: fmt.Sprintf("*Project*: <%v/det/projects/%v | %v>",
+				webUIBaseURL, projectID, pName),
+		})
+	} else if pName != "" {
+		expBlockFields = append(expBlockFields, SlackField{
+			Type: "mrkdwn",
+			Text: fmt.Sprintf("*Project*: %v", pName),
+		})
+	}
+	experimentBlock := SlackBlock{
+		Text: SlackField{
+			Type: "mrkdwn",
+			Text: eURL,
+		},
+		Type:   "section",
+		Fields: &expBlockFields,
+	}
+	messageBlock := SlackBlock{
+		Text: SlackField{
+			Text: status,
+			Type: "plain_text",
+		},
+		Type: "section",
+	}
+	attachment := SlackAttachment{
+		Color:  c,
+		Blocks: []SlackBlock{experimentBlock},
+	}
+	messageBody := SlackMessageBody{
+		Blocks:      []SlackBlock{messageBlock},
+		Attachments: &[]SlackAttachment{attachment},
+	}
+
+	message, err := json.Marshal(messageBody)
+	if err != nil {
+		return nil, fmt.Errorf("error creating slack payload: %w", err)
+	}
+	return message, nil
+}
+
+type eventBatch struct {
+	tx       *bun.Tx
+	events   []Event
+	consumed bool
```

Comment:
this flag may not be necessary. e.g. we often have a pattern like this for rollback:

```
	defer func() {
		if err = tx.Rollback(); err != nil && err != sql.ErrTxDone {
			log.WithError(err).Error("error rolling back transaction in create workspace")
		}
	}()
```
this way, the fact if the tx has been committed or now will be utilized instead.